pages/partials/submission.ejs

```diff
@@ -1,15 +1,38 @@
 <!-- Keeping "graded" state in the DOM is bad, yet here we are. -->
-<div class="card mb-4 pastsubmission-block" data-grading-job-status="<%= submission.grading_job_status %>" data-grading-job-id="<%= submission.grading_job_id %>" id="submission-<%= submission.id %>">
-  <div class="card-header bg-light text-dark d-flex align-items-center submission-header <% if (submissionIdx != submissionCount) { %> collapsed <% } %>" data-toggle="collapse" data-target="#submission-<%= submission.id %>-body">
+<div data-grading-job-status="<%= submission.grading_job_status %>" data-grading-job-id="<%= submission.grading_job_id %>" id="submission-<%= submission.id %>">
+
+<% if (course.manual_grading_visible && submission.feedback?.manual) { %>
+<div class="card mb-4 grading-block border-danger">
+  <div class="card-header bg-danger text-white d-flex submission-header <% if (submission.submission_number != submissionCount) { %> collapsed<% } %>" data-toggle="collapse" data-target="#submission-feedback-<%= submission.id %>-body">
+    <div class="mr-auto">
+      Grader feedback
+      <% if (submissionCount > 1) { %>
+      (for submitted answer <%= submission.submission_number %>)
+      <% } %>
+    </div>
+    <button type="button" class="expand-icon-container btn btn-outline-light btn-sm">
+      <i class="fa fa-angle-up fa-fw ml-1 expand-icon"></i>
+    </button>
+  </div>
+  <div class="collapse <% if (submission.submission_number == submissionCount) { %>show<% } %>" id="submission-feedback-<%= submission.id %>-body">
+    <div class="card-body feedback-body">
+      <%- submission.feedback_manual_html %>
+    </div>
+  </div>
+</div>
+<% } %>
+
+<div class="card mb-4 pastsubmission-block">
+  <div class="card-header bg-light text-dark d-flex align-items-center submission-header <% if (submission.submission_number != submissionCount) { %> collapsed <% } %>" data-toggle="collapse" data-target="#submission-<%= submission.id %>-body">
     <div class="mr-auto">
       <div>
         <span class="mr-2">
           Submitted answer
           <% if (submissionCount > 1) { %>
-          <%= submissionIdx %>
+          <%= submission.submission_number %>
           <% } %>
         </span>
-        <span><%- include('submissionStatus', {submission: submission}); %></span>
+        <span class="submission-status"><%- include('submissionStatus', {submission: submission}); %></span>
```

Comment:
If something is strictly to be used for testing, I'd rather use a `data-*` attribute that makes it really clear exactly what it is and what it's being used for:

```suggestion
        <span data-testid="submission-status"><%- include('submissionStatus', {submission: submission}); %></span>
```

You can then query this in the test with `$('[data-testid="submission-status"]')`. Check out this section of this blog post that talks about why this is better than classnames: https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change#whats-with-the-data-testid-query

Reply:
I have changed only the objects I created/modified in this PR, this is probably something to be added in a documentation or something like that for future items (and to eventually update in existing tests).